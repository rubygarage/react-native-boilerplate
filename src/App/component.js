import React from 'react';
import { Provider } from 'react-redux';
import { PersistGate } from 'redux-persist/integration/react';
import { IntlProvider } from 'react-intl-redux';

import configureStore from 'state/configureStore';
import AppModule from './AppModule';

window.navigator.userAgent = 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_2) AppleWebKit/537.36 (KHTML, like Gecko) ReactNativeDebugger/0.10.5 Chrome/78.0.3904.113 Electron/7.1.2 Safari/537.36';

const { store, persistor } = configureStore();

const App = () => (
  <Provider store={store}>
    <PersistGate loading={null} persistor={persistor}>
      <IntlProvider>
        <AppModule />
      </IntlProvider>
    </PersistGate>
  </Provider>
);

export default App;
