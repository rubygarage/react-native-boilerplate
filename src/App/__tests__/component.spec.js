import React from 'react';
import { shallow } from 'enzyme';

import App from '../component';

jest.useFakeTimers();

describe('App component', () => {
  const component = shallow(<App />);
  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });
});
