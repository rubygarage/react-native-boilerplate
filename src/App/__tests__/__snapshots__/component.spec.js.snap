// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "httpClient": [Function],
      "logicMiddleware": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <PersistGate
    loading={null}
    persistor={
      Object {
        "@@observable": [Function],
        "dispatch": [Function],
        "flush": [Function],
        "getState": [Function],
        "pause": [Function],
        "persist": [Function],
        "purge": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  >
    <Connect(IntlProvider)>
      <Connect(AppModule) />
    </Connect(IntlProvider)>
  </PersistGate>
</Provider>
`;
